server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -42,31 +42,21 @@ public class RemoteStoreNodeService {
     private final Supplier<RepositoriesService> repositoriesService;
     private final ThreadPool threadPool;
     public static final Setting<CompatibilityMode> REMOTE_STORE_COMPATIBILITY_MODE_SETTING = new Setting<>(
-        "remote_store.compatibility_mode",
+        "cluster.remote_store.compatibility_mode",
         CompatibilityMode.STRICT.name(),
         CompatibilityMode::parseString,
-        value -> {
-            if (value == CompatibilityMode.MIXED
-                && FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE_MIGRATION_EXPERIMENTAL_SETTING) == false) {
-                throw new IllegalArgumentException(
-                    " mixed mode is under an experimental feature and can be activated only by enabling "
-                        + REMOTE_STORE_MIGRATION_EXPERIMENTAL
-                        + " feature flag in the JVM options "
-                );
-            }
-        },
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
     public static final Setting<Direction> MIGRATION_DIRECTION_SETTING = new Setting<>(
-        "migration.direction",
+        "cluster.migration.direction",
         Direction.NONE.name(),
         Direction::parseString,
         value -> {
-            if (value != Direction.NONE && FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE_MIGRATION_EXPERIMENTAL_SETTING) == false) {
+            if (value == Direction.DOCREP && FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE_MIGRATION_EXPERIMENTAL_SETTING) == false) {
```

Comment:
If value is `DOCREP` and feature flag is enabled, it will not actually migrate, right?

Reply:
Yes, the feature to migrate to docrep is not complete .

Reply:
In that case, why don't we throw exception when value id DOCREP. We can allow the setting once the basic implementation of Remote Store to DocRep migration is done.

Reply:
In that case, we would need to remove experimental flag as well . That is why i prefer to have this as we want to support reverse migration as well later .